macros/src/serializable.rs

```diff
@@ -67,13 +53,67 @@ pub(crate) fn impl_derive_serializable(item: TokenStream) -> TokenStream {
         }
     };
 
-    let where_clause = if generics.params.is_empty() {
+    // Remove any bounds from the generic types and store them separately.
+    // Otherwise, collect_types will be called like `Foo::<T: MyTrait>::collect_types()` and where clauses
+    // will be incorrect, too.
+    let mut bounds = HashMap::new();
```

Comment:
Nitpick: Maybe a `BTreeMap` could be better here, in order to guarantee a reproducible order of the generated bounds.

Reply:
Good point!